### PR TITLE
Dockerfile: based on alpine

### PR DESCRIPTION
**Pull request by bryanlatten**
Ubuntu-based: 321.3MB
Alpine-based: 53.03MB

### CODE REVIEW

**Comment by Attamusc** [comment]
:+1:

**Comment by Attamusc** [comment]
every thumb in the world

**Comment by chr0n1x** [comment]
holy. hell.

**Comment by chr0n1x** [comment]
:+1:

File: container/root/etc/nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    #keepalive_timeout  0;
+    keepalive_timeout  65;
+
+    #gzip  on;
+
+    client_body_temp_path /tmp/client_body;
```
Comment (bryanlatten) [inline container/root/etc/nginx/nginx.conf]: setup for non-root nginx master process. still a few more tweaks to make before it works correctly

**Comment by anshupande** [comment]
neat alpine :) :+1:

**Comment by hzoo** [comment]
:tada:

File: container/root/etc/nginx/nginx.conf (hunk 1 above)
Comment (anshupande) [inline container/root/etc/nginx/nginx.conf]: :+1:

Comment (Attamusc) [inline container/root/etc/nginx/nginx.conf]: who cares it's cool lolkthxbro

**Comment by jimdelois** [comment]
:racehorse:

**Comment by bryanlatten** [comment]
@hzoo lets see what this does to the node `cruft` when it gets there.

File: Dockerfile [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
+        nginx \
+    && \
+    rm -rf /var/cache/apk/*
 
 # Overlay the root filesystem from this repo
 COPY ./container/root /
 
 EXPOSE 80
-CMD ["/bin/bash", "/run.sh"]
+CMD ["/bin/sh", "/run.sh"]
```
Comment (ko-be) [inline Dockerfile]: @bryanlatten we'll need to make sure we aren't using any bash-specific scripts anywhere in this container.

File: container/root/etc/nginx/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    '"$http_user_agent" "$http_x_forwarded_for" NGINX_SERVER';
+
+    error_log /dev/stdout info;
+    access_log /dev/stdout main;
+
+    sendfile        on;
+    #tcp_nopush     on;
+
+    #keepalive_timeout  0;
+    keepalive_timeout  65;
```
Comment (ko-be) [inline container/root/etc/nginx/nginx.conf]: This is the proper keepalive timeout for all nginx containers?  @bryanlatten

File: container/root/etc/nginx/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+# add to the run.d/nginx script
+#############################################################
+
+user nobody;
+worker_processes auto;
+
+pid /tmp/nginx.pid;
+
+events {
+    worker_connections 1024;
```
Comment (ko-be) [inline container/root/etc/nginx/nginx.conf]: Are there any filehandle limits, etc. we need to worry about w/ Alpine @bryanlatten ?

**Comment by ko-be** [comment]
A few comments, otherwise :+1:

**Comment by bossjones** [comment]
:+1:

**Comment by bossjones** [comment]
(this is amazing btw)

File: container/root/etc/nginx/nginx.conf (hunk 4 above)
Comment (bryanlatten) [inline container/root/etc/nginx/nginx.conf]: http://stackoverflow.com/questions/24318543/how-to-set-ulimit-file-descriptor-on-docker-container-the-image-tag-is-phusion

The file limits are actually from the host

**Pull request merged**